admin/includes/init_includes/init_aidba_install.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+if (!defined('IS_ADMIN_FLAG')) {
+    die('Illegal Access');
+}
+if(!defined('ADDITIONAL_IMAGES_APPROACH')) {
+    $db->Execute("INSERT INTO " . TABLE_CONFIGURATION . " (configuration_title, configuration_key, configuration_value, configuration_description, configuration_group_id, sort_order, set_function, date_added) VALUES ('Additional Images Approach', 'ADDITIONAL_IMAGES_APPROACH', 'old', 'Enable Additional Images Module', '4', '100', 'zen_cfg_select_option(array(\'old\', \'modern\'), ', NOW())");
+}
+
+if(!$sniffer->table_exists(TABLE_PRODUCTS_ADDITIONAL_IMAGES) && defined('ADDITIONAL_IMAGES_APPROACH') && ADDITIONAL_IMAGES_APPROACH == 'modern') {
+    // alter product table to InnoDB if not already done
+    $db->Execute("ALTER TABLE " . TABLE_PRODUCTS ." ENGINE=InnoDB;");
+
+    // create products_additional_images table
+    $db->Execute("CREATE TABLE " . TABLE_PRODUCTS_ADDITIONAL_IMAGES . " (
+            id INT(11) NOT NULL AUTO_INCREMENT PRIMARY KEY,
+            products_id INT(11) NOT NULL,
+            additional_image VARCHAR(255) NOT NULL,
+            sort_order INT(11) DEFAULT 0,
+            created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+            FOREIGN KEY (products_id) REFERENCES " . TABLE_PRODUCTS ."(products_id) ON DELETE CASCADE
+            ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;");
+
+    zen_register_admin_page('toolsAidba', 'BOX_TOOLS_CONVERT_AIDBA', 'FILENAME_AIDBA', '', 'tools', 'Y');
+}
```

Comment:
Since this is going into core, shouldn't this initialization be part of the zc_install for zc220?  The `CHARSET` specification on the create-table SQL should be dropped &mdash; while *all* stores should have their databases set to use a utf8mb4 variant by default, having the specification on the SQL could result in mixed collations for stores that are behind-the-times.

Q: Why the forcing of the database engine to InnoDB?

Reply:
uhm... to be honest, I'm not sure how to get it into the installer. Sorry.
But now that you mentioned it, we should probably move the extra definitions into the core files as well...

InnoDB - because of ON DELETE CASCADE. It requires InnoDB, and is a pretty safe way to ensure additional images are deleted from the database when a product is deleted so you don't have orphaned records.

Reply:
Hmm, I'm not sure how to get it into the installer either, since the `REFERENCES` needs to include the table-prefix if set.  Perhaps @drbyte can chime in on this.

Reply:
I'm stuck on the converter - as mentioned earlier by drbyte, not sure if it should be moved into an encapsulated plugin. If so, it might be quite counter-intuitive for the user to switch to the new style and then has to install the converter - this way it installs automatically and is straight-forward. On the other hand it is pretty much one-time use (unless it times out or fails so you need to re-run) so there's no point in leaving it in the Tools menu... Any suggestions?

Reply:
Sneaky, but ...

Upon successful completion, the tool could change its `admin_pages::display_on_menu` value to 'N'.  That way, the tool is still accessible via direct browser link but doesn't display anymore within the menu structure.

Reply:
Agreed. Was thinking of this as a bulletproof method of not having orphaned records in some wild case scenarios, but if it causes more trouble than it's potentially worth, it's better to just stick with the standard stuff. Will make the change (also noting that the `zen_remove_product` function needs a minor update as it will fail on delete from products_additional_images if still in legacy mode)

Reply:
If the `products_additional_images` table is **always** present, then I'm not seeing how the legacy vs. modern mode would affect that failure.

Noting that once the install is converted to a zc_install, that table's going to "always" be present.

Reply:
If you're in agreement to stick with the "standard stuff", I'll send you the zc_install `.sql` updates to your direct email address.

Reply:
And yes, of course I'm in agreement to stick with whatever is deemed best. You're more than welcome to email me, or just add the command yourself, whatever you feel is easier (my vote is on the latter).
I've updated the installer to this:
`$db->Execute("CREATE TABLE " . TABLE_PRODUCTS_ADDITIONAL_IMAGES . " (
            id INT(11) NOT NULL AUTO_INCREMENT PRIMARY KEY,
            products_id INT(11) NOT NULL,
            additional_image VARCHAR(255) NOT NULL,
            sort_order INT(11) DEFAULT 0,
            created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
            INDEX idx_products_id (products_id)
            ) ENGINE=MyISAM;");`
switched to MyISAM and added an index. Hopefully that would do it.